### PR TITLE
Dan

### PR DESCRIPTION
**Pull request by latos**
Please see the individual commits for the (several unrelated) changes

### CODE REVIEW

**Comment by latos** [comment]
@blambo

File: src/point.js [hunk 1]
```diff
@@ -238,15 +238,23 @@ Point.prototype.ensureInsertable = function() {
 Point.prototype.insert = function(newChild) {
   var p = this.rightNormalized();
 
+  p.ensureInsertable();
+
   if (p.type === TEXT) {
-    parentNode(p.node).insertBefore(newChild, p.node.splitText(p.offset));
-  } else if (p.type === BEFORE) {
+    p = p.moveToAfter(p.node).rightNormalized();
```
Comment (blambo) [inline src/point.js]: If we're inserting when a point is in a text node, why don't we want to split the text node?

File: src/point.js [hunk 2]
```diff
@@ -712,6 +712,82 @@ Point.prototype.setTo = function(point) {
...
+}
+
+function renderDebugNode(node, point) {
+  var isThisNode = (point.node === node);
+
+  var bits = [];
+
+  if (isThisNode && point.type === BEFORE) {
+    bits.push('\u2192'); // right arrow
+    //bits.push('[');
```
Comment (blambo) [inline src/point.js:734]: Scrap the commented out symbols

File: src/point.js [hunk 3]
```diff
@@ -712,6 +712,82 @@ Point.prototype.setTo = function(point) {
...
+  }
+  if (node.nodeType === 1) {
+    bits.push(renderDebugElem(node, point));
+  } else if (node.nodeType === 3) {
+    bits.push(renderDebugTextNode(node, point));
+  } else {
+    // TODO other stuff
```
Comment (blambo) [inline src/point.js:741]: Minor, but probably want more descriptive TODO

**Comment by blambo** [comment]
@latos One question for insert update. The other two comments are pretty minor.

**Comment by blambo** [comment]
Moving commits into ben branch, the question was the only blocking comment but have deleted as discussed in person as I was mistaken.

Will add a tidy up fixup commit that can get merged into ab8122e when ben branch is pushed across to master.

**Pull request merged**